kmeta/names_test.go

```diff
@@ -56,11 +58,25 @@ func TestChildName(t *testing.T) {
 		parent: "aaaa",
 		suffix: strings.Repeat("b---a", 20),
 		want:   "aaaa7a3f7966594e3f0849720eced8212c18b---ab---ab---ab---ab---ab",
-	}}
+	}, {
+		parent: strings.Repeat("a", 17),
+		suffix: "a.-.-.-.-.-.-.aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+		want:   "aaaaaaaaaaaaaaaaaa1eb10dc911444f8434af83b7225442da",
+	}, {
+		parent: strings.Repeat("A", 17),
+		suffix: "A.-.-.-.-.-.-.AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
```

Comment:
I think kubernetes resource names (which is implicitly the input to this function) will already be forced to lowercase.

I had meant that the _input_ to the function was constrained, and the lack of capital letters in the test was evidence of this.

Sorry for the confusion!

Reply:
no worries, thanks for clarifying!